### PR TITLE
Add Consistent `bands` Metadata to Vision Transformer and ResNet Weights

### LINKED ISSUE #2364: Missing Band information for pretrained weights
### Issue

The [satlas weights](https://github.com/microsoft/torchgeo/blob/67fe198c78128e1162a241a783ac20ce350b52db/torchgeo/models/resnet.py#L527) have a `bands` entry in the metadata that lists the order and bands, for consistency and common api, it would be helpful to have that same metadata entry for all the other weights as well.

### Fix

Add `bands` info to metadata.

@paolofraccaro

### PR DESCRIPTION
**Pull request by InderParmar**
This PR introduces consistent `bands` metadata across `vit.py` and `resnet.py` weight classes to enhance usability and ensure alignment with existing metadata standards in the project.

#### Changes to `vit.py`
- Added `bands` metadata for Vision Transformer (ViT) weight classes.
- Updated `vit.py` to import `_satlas_bands`, `_satlas_sentinel2_bands`, and `_satlas_landsat_bands` from `swin.py` for consistent band definitions.
- Applied correct band metadata for each ViT weight class, supporting Landsat and Sentinel datasets, ensuring metadata aligns with project standards.

#### Changes to `resnet.py`
- Added `bands` metadata for ResNet weight classes, providing consistent metadata across ResNet weights.
- Updated `resnet.py` to import `_satlas_bands`, `_satlas_sentinel2_bands`, and `_satlas_landsat_bands` from `swin.py`.
- Updated metadata fields to align with the Sentinel and Landsat dataset specifications, improving compatibility and usability for model users.
Fixes #2364

### CODE REVIEW

**Comment by InderParmar** [comment]
> @InderParmar please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
>  > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement



> @InderParmar please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
>  > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

File: torchgeo/models/resnet.py [hunk 1]
```diff
@@ -138,6 +139,7 @@ class ResNet18_Weights(WeightsEnum):  # type: ignore[misc]
             'publication': 'https://arxiv.org/abs/2306.09424',
             'repo': 'https://github.com/microsoft/torchgeo',
             'ssl_method': 'moco',
+            'bands': _satlas_landsat_bands,
```
Comment (adamjstewart) [inline torchgeo/models/resnet.py:142]: `_satlas_landsat_bands` contains 11 bands, but this model was only trained on 7 (see `in_chans`). None of these models are Satlas models, so there's no reason to believe they should have the same bands as Satlas.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: torchgeo/models/resnet.py (hunk 1 above)
Comment (InderParmar) [reply torchgeo/models/resnet.py]: > `_satlas_landsat_bands` contains 11 bands, but this model was only trained on 7 (see `in_chans`). None of these models are Satlas models, so there's no reason to believe they should have the same bands as Satlas.

@adamjstewart , Hi, sorry for the confusion on my part, now I have made the appropriate changes, can you review the PR, and if there are no more concerns, can you add the "hacktoberfest-accepted" label to the PR, and then merge it before 1 November, I will be very thankful.

**Comment by InderParmar** [comment]
Hi @adamjstewart , sorry for the confusion on my part, now I have made the appropriate changes, can you review the PR, and if there are no more concerns, can you add the "hacktoberfest-accepted" label to the PR, and then merge it before 1 November, I will be very thankful.

File: torchgeo/models/resnet.py [hunk 2]
```diff
@@ -18,6 +18,92 @@
     _satlas_transforms,
 )
 
+# Landsat TM-toa 7 channels
+bands_7_channels_tm_toa = [
```
Comment (adamjstewart) [inline torchgeo/models/resnet.py:22]: I would name this something like `_landsat_tm_toa_bands`, similar for other variables

File: torchgeo/models/resnet.py [hunk 3]
```diff
@@ -18,6 +18,92 @@
...
+    'B1',
+    'B2',
+    'B3',
+    'B4',
+    'B5',
+    'B6',
+    'B7',
+]  # https://developers.google.com/earth-engine/datasets/catalog/LANDSAT_LT05_C02_T1_TOA
```
Comment (adamjstewart) [inline torchgeo/models/resnet.py:30]: Can we move this comment to above the variable (so line 22)? Similar for others. This will allow more lists to be displayed on a single line

File: torchgeo/models/resnet.py [hunk 4]
```diff
@@ -18,6 +18,92 @@
     _satlas_transforms,
 )
 
+# Landsat TM-toa 7 channels
```
Comment (adamjstewart) [inline torchgeo/models/resnet.py:21]: TOA (top of atmosphere) and SR (surface reflectance) should be capitalized

**Comment by InderParmar** [comment]
Hi @adamjstewart , I have made all the requested changes, can you please review the PR, and let me know if there are any further changes required. Looking forward to it,If you find the PR is good, please add the "hacktoberfest-accepted" label to it before merging.

**Comment by adamjstewart** [comment]
I ended up removing the fMoW band designations. The RGB version of the dataset was created using a combination of:

| # bands | satellites | RGB channels |
| --- | --- | --- |
| 4 | [QuickBird-2](https://earth.esa.int/eogateway/missions/quickbird-2), [GeoEye-1](https://earth.esa.int/eogateway/missions/geoeye-1) | 3, 2, 1 |
| 8 | [WorldView-2](https://earth.esa.int/eogateway/missions/worldview-2), [WorldView-3](https://earth.esa.int/eogateway/missions/worldview-3) | 5, 3, 2 |

Since the band numbers are not consistent, there's no way to provide a single set of band indices.

**Review by adamjstewart — APPROVED** [review]
